### PR TITLE
Remove the freeimage plugin

### PR DESCRIPTION
**Pull request by blink1073**
As discussed in #1463.

### CODE REVIEW

**Comment by soupault** [comment]
xref https://github.com/scikit-image/scikit-image/issues/1195

File: doc/source/api_changes.txt [hunk 1]
```diff
@@ -4,6 +4,8 @@ Version 0.12
   the ``ntiles_*`` arguments.
 - The functions ``blob_dog``, ``blob_log`` and ``blob_doh`` now return float
   arrays instead of integer arrays.
+- The ``freeimage`` io plugin is no longer supported, and will use ``imageio``
+instead.
```
Comment (soupault) [inline doc/source/api_changes.txt]: Should have the same indent as above to pass Travis.

File: skimage/io/tests/test_freeimage.py [hunk 2]
```diff
@@ -1,129 +1,65 @@
...
-@skipif(not FI_available)
-def test_imread_truncated_jpg():
-    assert_raises((RuntimeError, ValueError),
-                  sio.imread,
-                  os.path.join(si.data_dir, 'truncated.jpg'))
-
-@skipif(not FI_available)
-def test_imread_uint16():
-    expected = np.load(os.path.join(si.data_dir, 'chessboard_GRAY_U8.npy'))
-    img = sio.imread(os.path.join(si.data_dir, 'chessboard_GRAY_U16.tif'))
```
Comment (stefanv) [inline skimage/io/tests/test_freeimage.py]: Are there any other tests / functions that use chessboard_GRAY_U16?

**Comment by stefanv** [comment]
We should have a way to encourage users to install ``imageio``.  In the docs?  Listed under optional dependencies?  In a warning raised when no suitable plugin can be found?  As a hard dependency?

File: skimage/io/tests/test_freeimage.py (hunk 2 above)
Comment (soupault) [inline skimage/io/tests/test_freeimage.py]: grep indicates also:
```
./skimage/io/tests/test_simpleitk.py
./skimage/io/tests/test_pil.py
./skimage/io/tests/test_tifffile.py
```

Comment (stefanv) [inline skimage/io/tests/test_freeimage.py]: OK, good, then we don't have to remove them.

**Comment by stefanv** [comment]
Please see my comment on https://github.com/imageio/imageio/issues/114.  We may have to also remove the dependency on imageio due to freeimage licensing problems.

**Comment by stefanv** [comment]
@blink1073  ``imageio`` ships several binaries (such as freeimage) inside of their wheels.

**Comment by stefanv** [comment]
Note that ImageIO has now removed freeimage as the default plugin, instead using PIL.  This paves the way forward for skimage to use imageio as its default I/O backend!

**Comment by blink1073** [comment]
Great news!

**Comment by JDWarner** [comment]
:+1:

File: skimage/io/tests/test_freeimage.py [hunk 3]
```diff
@@ -1,129 +1,65 @@
...
-    x = np.ones(shape, dtype=np.uint8) * np.random.rand(*shape) * 255
-    x = x.astype(np.uint8)
-    f = NamedTemporaryFile(suffix='.tif')
-    fname = f.name
-    f.close()
-    fi.write_multipage(x, fname)
-    y = fi.read_multipage(fname)
-    assert_array_equal(x, y)
+@skipif(not imageio_available)
+def test_freeimage_flatten():
```
Comment (soupault) [inline skimage/io/tests/test_freeimage.py]: Any reasons to keep this?

Comment (blink1073) [inline skimage/io/tests/test_freeimage.py]: Nope, let's pull it.

File: doc/source/api_changes.rst [hunk 4]
```diff
@@ -17,6 +17,8 @@ Version 0.13
   0.15, will be set to `True`.
 - Parameters ``ntiles_x``, ``ntiles_y`` have been removed from
   ``skimage.exposure.equalize_adapthist``.
+- The ``freeimage`` io plugin is no longer supported, and will use ``imageio``
```
Comment (sciunto) [inline doc/source/api_changes.rst]: There is something I misunderstand. You removed a task attributed for 0.14 and you added an item for 0.13. When do we want to remove freeimage?

We need to agree on that and then, be consistent.

**Comment by blink1073** [comment]
I added a clarifying note.  This PR removes the local implementation of the freeimage plugin and instead shims the imageio one.  The todo now reflects that we should remove that shim in 0.14.

**Comment by sciunto** [comment]
LGTM! Thanks. We need a second review.

File: doc/source/api_changes.rst (hunk 4 above)
Comment (soupault) [inline doc/source/api_changes.rst]: As long as this goes to release notes, should we add a sentence that the plugin will be completely removed in 0.14?

Comment (blink1073) [inline doc/source/api_changes.rst]: Done.

**Comment by soupault** [comment]
Let's move forward.
Thanks for the work, Steven!

**Pull request merged**